server/subscribe.go

```diff
@@ -114,6 +114,61 @@ func (p *Plugin) getUserID() string {
 	return p.getConfig().botUserID
 }
 
+func (p *Plugin) matchesSubsciptionFilters(wh *webhook, filters SubscriptionFilters) bool {
+	webhookEvents := wh.Events()
+	foundEvent := false
+	eventTypes := filters.Events
+	if eventTypes.Intersection(webhookEvents).Len() > 0 {
+		foundEvent = true
+	} else if eventTypes.ContainsAny(eventUpdatedAny) {
+		for _, eventType := range webhookEvents.Elems() {
+			if strings.HasPrefix(eventType, "event_updated") {
+				foundEvent = true
+			}
+		}
+	}
+
+	if !foundEvent {
+		return false
+	}
+
+	if filters.IssueTypes.Len() != 0 && !filters.IssueTypes.ContainsAny(wh.JiraWebhook.Issue.Fields.Type.ID) {
```

Comment:
The only change in the refactor here is this line and the similar line for Projects below. (see description for details)

Reply:
Can you please explain why it was related? (Can't figure it out)

Reply:
I did this to make it easier to specify filters in the workflow JSON. Instead of having to specify every issue type, you can just use "" for all.

Reply:
Sounds good to me. The issue type and event ones seem like something that may be useful for the UI too. So you don't have to select every single one every time if you want all of them.

@aaronrothschild Thoughts on this? Users would be able to leave "Issue Type" and/or "Events" blank in the subscription modal, which would translate to "All Issue Types" and "All Events".

Reply:
1/5 but the semantics of how we represent it aside, any long list of individually selectable items should IMO include a "Select All" capability, and then un-select what one doesn't want. I got personally annoyed by this when I was setting up subscriptions on community, but might have been a different field that annoyed me.

Reply:
cc @asaadmahmood